pkg/controller/infinispan/resources/operatorconfig/grafana.go

```diff
@@ -0,0 +1,95 @@
+package operatorconfig
+
+import (
+	"github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	consts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	kube "github.com/infinispan/infinispan-operator/pkg/kubernetes"
+	grafanav1alpha1 "github.com/integr8ly/grafana-operator/v3/pkg/apis/integreatly/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	grafanaDashboardNamespace = "grafana.dashboard.namespace"
+)
+
+// reconcileGrafana reconciles grafana object status with the operator configuration settings
+func (r *ReconcileOperatorConfig) reconcileGrafana(config map[string]string, currentConfig map[string]string, operatorNs string) (*reconcile.Result, error) {
+	grafanaNs := config[grafanaDashboardNamespace]
+	currGrafanaNs := currentConfig[grafanaDashboardNamespace]
+	// Delete current grafana dashboard if namespace is changes
+	if grafanaNs != currGrafanaNs && currGrafanaNs != "" {
+		currGrafana := &grafanav1alpha1.GrafanaDashboard{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.GetEnvWithDefault(ApplicationNameVariable, DefaultApplicationName),
```

Comment:
Is it possible to store all required fields inside config. Why we need to mix evv variables and ConfigMap values?

Reply:
This sound really good, thanks!